nengo/synapses.py

```diff
@@ -119,24 +161,53 @@ def filtfilt(self, x, **kwargs):
         """
         return self.filt(x, filtfilt=True, **kwargs)
 
+    def _sample_initial_output(self, shape, y0=None, rng=np.random):
+        if self.initial_output is None and y0 is None:
+            return None
+
+        output = self.initial_output if y0 is None else y0
+        if isinstance(output, Distribution):
+            output = output.sample(n=np.prod(shape), d=1).reshape(shape)
+        else:
+            output = np.array(output, copy=False, ndmin=len(shape))
+
+        assert len(output.shape) >= len(shape)
+        too_long = len(output.shape) > len(shape)
+        if too_long or not all(o in (1, s) for o, s in zip(output.shape, shape)):
+            details = " (output shape is too long)" if too_long else ""
+            raise ValidationError(
+                "Output shape %s is not broadcastable to filter shape %s%s"
+                % (output.shape, shape, details),
+                "initial_output",
+                obj=self,
+            )
+
+        return output
 
-class LinearFilter(Synapse):
+
+class LinearFilter(LinearSystem, Synapse):
     """General linear time-invariant (LTI) system synapse.
 
-    This class can be used to implement any linear filter, given the
-    filter's transfer function. [1]_
+    This class can be used to implement any linear filter, given a description of the
+    filter. [1]_
 
     Parameters
     ----------
-    num : array_like
-        Numerator coefficients of transfer function.
-    den : array_like
-        Denominator coefficients of transfer function.
+    sys : tuple
+        A tuple describing the linear system used for filtering, in
+        ``(numerator, denominator)``, ``(zeros, poles, gain)``, or ``(A, B, C, D)``
+        (state space) form. See `.LinearSystem` for more details on these forms.
+        The system must have both an input size and output size of 1.
+    den : array_like, optional (DEPRECATED)
+        Denominator coefficients of the transfer function. If used, the first (``sys``)
+        argument is treated as the numerator coefficients of the transfer function.
+        This way of specifying transfer functions is deprecated; the preferred method
+        is to pass a ``(num, den)`` tuple to the ``sys`` argument.
```

Comment:
Do these need to have any sort of `.. versionchanged` or `.. versiondeprecated` sphinx annotation or similar?

Reply:
Sure!